app/scripts/modules/amazon/src/pipeline/stages/deployCloudFormation/evaluateCloudFormationChangeSetExecutionMarkerIcon.tsx

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { IExecutionMarkerIconProps } from 'core/pipeline/config/stages/common/ExecutionMarkerIcon';
+
+export class EvaluateCloudFormationChangeSetExecutionMarkerIcon extends React.Component<IExecutionMarkerIconProps> {
+  constructor(props: IExecutionMarkerIconProps) {
+    super(props);
+  }
+
+  public render() {
+    if (
+      this.props.stage.isRunning &&
+      this.props.stage.stages[0].context.changeSetContainsReplacement &&
+      this.props.stage.stages[0].context.actionOnReplacement === 'ask'
+    ) {
+      require('./evaluateCloudFormationChangeSetExecutionDetailsAsk.less');
+      if (
+        typeof document.getElementsByClassName('execution-marker-running stage-type-deploycloudformation')[0] !==
+        'undefined'
+      ) {
+        document
+          .getElementsByClassName('execution-marker-running stage-type-deploycloudformation')[0]
+          .classList.add('stage-type-deploycloudformation-ask');
```

Comment:
There is a package called `classnames` where you can construct an object with conditional classnames. It could replace this entire if block.

```
 const className = classNames({
      'class1': alwaysTrue,
      'stage-type-deploycloudformation-ask': whenThisIsTrue,
    });
```